packages/gatsby/src/commands/build-html.js

```diff
@@ -41,14 +41,16 @@ module.exports = async (program: any) => {
         )
       }
 
-      // Remove the temp JS bundle file built for the static-site-generator-plugin
-      try {
-        fs.unlinkSync(outputFile)
-        fs.unlinkSync(`${outputFile}.map`)
-      } catch (e) {
-        // This function will fail on Windows with no further consequences.
-      }
-      return resolve(null, stats)
+      return renderHTML(require(outputFile), pages).then(() => {
```

Comment:
will just requiring this always work? not sure how the various async imports or jsonp stuff will be handled...I guess this is what the plugin used to do tho

Reply:
this should work, as gatsby is in control of this - `outputFile` is compiled `.cache/static-entry.js` which sync requires all the templates (and for now layouts) - so any dynamic imports that user may be using are below that and would be handled same as they are currently